src/app/gateways/views/BuiltinGatewayDetailView.vue

```diff
@@ -18,29 +18,19 @@
       }"
     >
       <AppView>
-        <DataSource
+        <DataLoader
           v-slot="{ data, error }: MeshGatewaySource"
           :src="`/meshes/${route.params.mesh}/mesh-gateways/${route.params.gateway}`"
         >
-          <ErrorBlock
-            v-if="error"
-            :error="error"
-          />
-
-          <LoadingBlock v-else-if="data === undefined" />
-
-          <KCard v-else>
-            <DataSource
-              v-slot="{ data: dataplanesData, error: dataplanesError }: DataplaneOverviewCollectionSource"
-              :src="`/meshes/${route.params.mesh}/dataplanes/for/${data.selectors[0].match['kuma.io/service']}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"
-            >
-              <ErrorBlock
-                v-if="dataplanesError !== undefined"
-                :error="dataplanesError"
-              />
-
+          <DataLoader
+            v-if="data"
```

Comment:
FYI: I've been using `data!` inside `DataLoader`s for the moment. It should always be set inside the default slot, and once I finish up the DataSource URI/src auto typing it will end up being typed correctly. Kinda of a nit this so I guess up to you.

Reply:
I think the `v-if` directive is much more expressive in saying "use this content if you have data and you may access it in the following descendant nodes". Since I access `data` further down, this also saves me from constantly using `data!`.

Reply:
cool I don't really mind either way at the min. Just to note the `v-if` is runtime, I don't believe the `!` is? FYI: one of the reasons for DataLoader working as it does is to minimize the need for conditionals everywhere (once the URIs are typed all neither of these approaches will be needed I don't think)

Gonna resolve this one.